main/options.c

```diff
@@ -597,7 +597,7 @@ static vString* getHome (void)
 	}
 }
 
-#if defined(_WIN32) && !(defined(__USE_MINGW_ANSI_STDIO) && defined(__MINGW64_VERSION_MAJOR))
+#if defined(_WIN32) && !(defined(__USE_MINGW_ANSI_STDIO) && defined(__MINGW64_VERSION_MAJOR)) || !defined(HAVE_ASPRINTF)
```

Comment:
I don't know what the Mingw stuff is there, but it seems to me this whole thing could be replaced with:

```c
#ifndef HAVE_ASPRINTF
```

Reply:
I assumed that at least windows is compiled without running configure, but I haven't deeply explored the buildsystem for that.
I decided to take a safe route here, as I'm not knowledgable enough to make a decision about the existing conditions.

Reply:
I haven't tested but this might be useful when `mk_mingw.mak` is used with MinGW-w64.
MinGW-w64 has `asprintf`, but `mk_mingw.mak` (and `src/e_msoft.h`) doesn't define `HAVE_ASPRINTF`.
(Moving the condition into `e_msoft.h` might be another option.)

Reply:
I mean, for example:

```diff
--- a/main/e_msoft.h
+++ b/main/e_msoft.h
@@ -66,6 +66,10 @@ typedef enum { false, true } bool;
 # define FA_DIREC _A_SUBDIR
 # define ff_name name
 
+# if defined(__USE_MINGW_ANSI_STDIO) && defined(__MINGW64_VERSION_MAJOR)
+#  define HAVE_ASPRINTF 1
+# endif
+
 #endif
 
 #endif
```
(Not tested)

Reply:
O.k. I will make a pull request based on @grobian with the comment of @codebrainz and @k-takata.

Reply:
(After fixing the trouble about CI I'm working for).

Reply:
Do you want me to try and put both suggestions in this PR and see if all tests turn green?

Reply:
@grobian, YES!
Could you rebase the lastest code? I've fixed bugs running test cases on CI environements.